RecoMET/METPUSubtraction/src/noPileUpMEtAuxFunctions.cc

```diff
@@ -18,20 +18,17 @@ int isVertexAssociated(const reco::PFCandidate& pfCandidate,
     vtxAssociationType = noPuUtils::kChNoAssoc;
     for ( PFCandToVertexAssMap::const_iterator pfCandToVertexAssociation = pfCandToVertexAssociations.begin();
           pfCandToVertexAssociation != pfCandToVertexAssociations.end(); ++pfCandToVertexAssociation ) {
-      typedef std::vector<std::pair<reco::PFCandidateRef, int> > PFCandidateQualityPairVector;
-      const PFCandidateQualityPairVector& pfCandidates_vertex = pfCandToVertexAssociation->val;
-      for ( PFCandidateQualityPairVector::const_iterator pfCandidate_vertex = pfCandidates_vertex.begin();
-            pfCandidate_vertex != pfCandidates_vertex.end(); ++pfCandidate_vertex ) {
-	int pfCandToVertexAssocQuality = pfCandidate_vertex->second;
-	if ( pfCandToVertexAssocQuality >= minPFCandToVertexAssocQuality && deltaR2(pfCandidate.p4(), pfCandidate_vertex->first->p4()) < dR2Min ) {
-          if ( vtxAssociationType < noPuUtils::kChPUAssoc ) vtxAssociationType = noPuUtils::kChPUAssoc;
-          for ( reco::VertexCollection::const_iterator vertex = vertices.begin();
-                vertex != vertices.end(); ++vertex ) {
-            if ( std::abs(pfCandToVertexAssociation->key->position().z() - vertex->position().z()) < dZ ) {
-              if ( vtxAssociationType < noPuUtils::kChHSAssoc ) vtxAssociationType = noPuUtils::kChHSAssoc;
-            }
-          }
-        }
+     
+      const noPuUtils::CandQualityPairVector& pfCandidates_vertex = pfCandToVertexAssociation->val;
+
+     
+      for ( noPuUtils::CandQualityPairVector::const_iterator pfCandidate_vertex = pfCandidates_vertex.begin();
+	    pfCandidate_vertex != pfCandidates_vertex.end(); ++pfCandidate_vertex ) {
+	 
+	if(deltaR2(pfCandidate.p4(), pfCandidate_vertex->first->p4()) > dR2Min ) continue;
```

Comment:
Actually this function is kept for potential debugging but is not used  in the algorithm, which uses the isVertexAssociated_fast function which makes in priority a key by key matching instead of a deltaR matching (done only if no key matching is obtained, but this is only relevant when uncertainties are computed)